src/main/java/game/Actions.java

```diff
@@ -25,12 +31,13 @@ public boolean onCommand(Command command, Callback callback) {
     }
 
     @Override
-    public List<String> listCommands(Game game, List<String> addToThisList) {
-        Set<String> commands = getCommands();
+    public List<String> listHandledCommands(Game game) {
+        List<String> result = new ArrayList<>();
+        Set<String> commands = getCommandSet();
         if (commands != null) {
-            addToThisList.addAll(getCommands());
+            result.addAll(getCommandSet());
```

Comment:
Here instead of calling getCommandSet() again, you could have used the commands set variable to which you have already assigned the result